src/components/SearchForm.js

```diff
@@ -171,56 +163,62 @@ const SearchForm = ({onFormChange}) => {
           <CharacterSelectButton
             name="characterRarity"
             label="UR"
-            onChange={setCharacterRarity}
-            currentSelect={characterRarity}
           />
           <CharacterSelectButton
             name="characterRarity"
             label="LR"
-            onChange={setCharacterRarity}
-            currentSelect={characterRarity}
           />
           <CharacterSelectButton
             name="characterRarity"
+            value=""
             label="ALL"
-            onChange={() => setCharacterRarity("")}
-            currentSelect={characterRarity}
+            defaultChecked
           />
         </div>
 
         <div className="order-5 bg-orange-300 rounded-md border-2 border-slate-900 flex 2xl:h-12 2xl:mt-5">
-          <button
-            className="pr-10 pl-10 pt-2 pb-2 relative hover:bg-orange-400 m-0.5"
+        <label
+          for="isUserDeck"
+        >
+          <input
+            type="checkbox"
             name="isUserDeck"
-            style={{
-              backgroundColor: isUserDeck ? "DarkOrange" : "Orange",
-            }}
-            onClick={() => setIsUserDeck((prev) => !prev)}
+            id="isUserDeck"
+            className="hidden peer"
+            value={true}
+          />
+          <div 
+            style={{cursor: "pointer"}}
+            className="pr-10 pl-10 pt-2 pb-2 relative hover:bg-orange-400 m-0.5 peer-checked:bg-orange-400"
           >
-            My Deck
-          </button>
+              My Deck
+            </div>
+        </label>
         </div>
       </form>
     </div>
   );
 };
 
-const CharacterSelectButton = ({ name, label, onChange, currentSelect }) => {
+const CharacterSelectButton = ({ name, label, ...inputProps}) => {
   return (
     <label
-      for={label}
-      style={{ appearance: "button" }}
-      className="pr-10 pl-10 pt-2 pb-2 relative hover:bg-orange-400 m-0.5"
+      for={`${name}-${label}`}
     >
       <input
+        type="radio"
         name={name}
-        id={label}
-        style={{
-          backgroundColor: currentSelect === label ? "DarkOrange" : "Orange",
-        }}
-        onClick={() => onChange(label)}
+        id={`${name}-${label}`}
+        className="hidden peer"
```

Comment:
hidden means the radio doesn't show so we just have the div below acting as the button